src/server/mod.rs

```diff
@@ -302,6 +302,17 @@ impl<I, E> Builder<I, E> {
         self
     }
 
+    /// Sets the [`SETTINGS_MAX_CONCURRENT_STREAMS`][spec] option for HTTP2
+    /// connections.
+    ///
+    /// Default is no limit (`None`).
+    ///
+    /// [spec]: https://http2.github.io/http2-spec/#SETTINGS_MAX_CONCURRENT_STREAMS
+    pub fn http2_max_concurrent_streams(mut self, max: impl Into<Option<u32>>) -> Self {
+        self.protocol.http2_max_concurrent_streams(max.into());
```

Comment:
Why do you have `max.into()` here, but

```
if let Some(max) = max.into {
    ...
}
```

in `src/server/conn`?

Reply:
Hm, didn't think much about it, really!

Reply:
Oh okay!

Reply:
Oh no this makes sense. `h2`'s `fn max_concurrent_streams` needs a value, while here you are passing in an `Option` to `conn::Builder`'s wrapper.